studio/src/pages/RequestorPage/RequestorPageContent/RequestorPageContent.tsx

```diff
@@ -32,12 +42,93 @@ type RequestorPageContentProps = {
   generateNavigation: (traceId: string) => To;
 };
 
+/**
+ * Gets the collection item for the current route (if applicable)
+ */
+function useCollectionItem():
+  | { collectionItem: CollectionItem; appRoute: ProbedRoute }
+  | undefined {
+  const collectionId = useActiveCollectionId();
+  const itemId = useActiveCollectionItemId();
+
+  const { collections, appRoutes } = useStudioStore("collections", "appRoutes");
+
+  if (itemId === null || collectionId === null) {
+    return;
+  }
+
+  const collection = collections.find((item) => item.id === collectionId);
+  const collectionItem = collection?.collectionItems.find(
+    (element) => element.id === itemId,
+  );
+  const appRoute = appRoutes.find(
+    (route) => route.id === collectionItem?.appRouteId,
+  );
+
+  if (!collectionItem || !appRoute) {
+    return;
+  }
+
+  return {
+    collectionItem,
+    appRoute,
+  };
+}
+
 export const RequestorPageContent: React.FC<RequestorPageContentProps> = (
   props,
 ) => {
   const { history, overrideTraceId, historyLoading, generateNavigation } =
     props;
 
+  const activeTraceId = useActiveTraceId();
+  const item = history.find((element) => getId(element) === activeTraceId);
+
+  const {
+    togglePanel,
+    setAIDropdownOpen,
+    setAiPrompt,
+    setRequestParams,
+    updateMethod,
+  } = useStudioStore(
+    "togglePanel",
+    "setAIDropdownOpen",
+    "setAiPrompt",
+    "setRequestParams",
+    "updateMethod",
+  );
+
+  const { collectionItem, appRoute } = useCollectionItem() ?? {};
+  const itemId = collectionItem?.id;
+  const collectionItemRef = useLatest<CollectionItem | undefined>(
+    collectionItem,
+  );
+  const appRouteRef = useLatest<ProbedRoute | undefined>(appRoute);
+
+  useEffect(() => {
+    if (itemId === null || !collectionItemRef.current || !appRouteRef.current) {
+      return;
+    }
+    const { id, appRouteId, ...extraParams } = collectionItemRef.current;
+
+    // const {}
+    setRequestParams({
+      ...extraParams,
+      requestUrl: `http://localhost:8787${appRouteRef.current.path}`,
+      requestMethod: appRouteRef.current.method,
+      requestRoute: appRouteId.toString(),
+    });
+    updateMethod(appRouteRef.current.method);
+  }, [itemId, collectionItemRef, appRouteRef, setRequestParams, updateMethod]);
+
+  useEffect(() => {
+    if (!item) {
+      return;
+    }
+
+    setRequestParams(item.app_requests);
+  }, [item, setRequestParams]);
+
```

Comment:
something is a little wonky with the introduction of these effects — i'm getting some flash-in behavior, especially where the UI will temporarily show the "activeResponse", then flash to the trace response. 

aside from that, the headers will temporarily flash to the list of headers made in the proxied request, then flash back to the og headers in the form. 

will try to resolve today

Reply:
at any rate, the effects could probably use comments that explain their purpose

Reply:
removing the second effect in this highlighted code did the trick for me. i'm not sure if it served a purpose, trying to look for things i might have broken